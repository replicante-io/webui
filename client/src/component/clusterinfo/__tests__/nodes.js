'use strict';
//@flow

import React from 'react';
import ReactTestUtils from 'react-dom/test-utils';
import { Provider } from 'react-redux';

import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';

import Nodes from '../nodes';


const mockStore = configureStore([]);
const MATCH = {
  params: {name: 'test'}
};


describe('ClusterInfo', () => {
  describe('nodes', () => {
    let store;

    beforeEach(() => {
      store = mockStore({
        clusterinfo: {
          nodes: {}
        }
      });
    });

    test('renders no data', () => {
      const tree = renderer.create(
        <Provider store={store}>
          <Nodes match={MATCH} />
        </Provider>
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });

    test('renders with data', () => {
      store = mockStore({ clusterinfo: { nodes: {
        'test': [{
          cluster_id: "replistore",
          kind: "MongoDB",
          node_id: "localhost:27017",
          version: "4.0.10",
        }]
      } } });
      const tree = renderer.create(
        <Provider store={store}>
          <Nodes match={MATCH} />
        </Provider>
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });

  });
});
