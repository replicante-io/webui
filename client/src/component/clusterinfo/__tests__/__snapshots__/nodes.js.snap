// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClusterInfo nodes renders no data 1`] = `
<div
  className="text-center"
>
  <p
    className="large-icon nodata"
  >
    <span
      className="fa-layers fa-fw"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-database "
        data-icon="database"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={{}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M448 80V128C448 172.2 347.7 208 224 208C100.3 208 0 172.2 0 128V80C0 35.82 100.3 0 224 0C347.7 0 448 35.82 448 80zM393.2 214.7C413.1 207.3 433.1 197.8 448 186.1V288C448 332.2 347.7 368 224 368C100.3 368 0 332.2 0 288V186.1C14.93 197.8 34.02 207.3 54.85 214.7C99.66 230.7 159.5 240 224 240C288.5 240 348.3 230.7 393.2 214.7V214.7zM54.85 374.7C99.66 390.7 159.5 400 224 400C288.5 400 348.3 390.7 393.2 374.7C413.1 367.3 433.1 357.8 448 346.1V432C448 476.2 347.7 512 224 512C100.3 512 0 476.2 0 432V346.1C14.93 357.8 34.02 367.3 54.85 374.7z"
          fill="currentColor"
          style={{}}
        />
      </svg>
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-question overlay"
        data-icon="question"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={{}}
        viewBox="0 0 320 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M204.3 32.01H96c-52.94 0-96 43.06-96 96c0 17.67 14.31 31.1 32 31.1s32-14.32 32-31.1c0-17.64 14.34-32 32-32h108.3C232.8 96.01 256 119.2 256 147.8c0 19.72-10.97 37.47-30.5 47.33L127.8 252.4C117.1 258.2 112 268.7 112 280v40c0 17.67 14.31 31.99 32 31.99s32-14.32 32-31.99V298.3L256 251.3c39.47-19.75 64-59.42 64-103.5C320 83.95 268.1 32.01 204.3 32.01zM144 400c-22.09 0-40 17.91-40 40s17.91 39.1 40 39.1s40-17.9 40-39.1S166.1 400 144 400z"
          fill="currentColor"
          style={{}}
        />
      </svg>
    </span>
  </p>
  No information available. 
  <br />
  If nothing shows up the cluster may not exist.
</div>
`;

exports[`ClusterInfo nodes renders with data 1`] = `
<table
  className="table"
>
  <thead>
    <tr>
      <th>
        Node ID
      </th>
      <th>
        Software Kind
      </th>
      <th>
        Version
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        localhost:27017
      </td>
      <td>
        MongoDB
      </td>
      <td>
        4.0.10
      </td>
    </tr>
  </tbody>
</table>
`;
